allure-generator/src/main/java/io/qameta/allure/history/HistoryTrendPlugin.java

```diff
@@ -78,22 +82,37 @@ public String getName() {
         return HISTORY_TREND;
     }
 
-    private List<Statistic> getHistoryTrendData(final List<LaunchResults> launchesResults) {
+    private List<HistoryTrendItem> getHistoryTrendData(final List<LaunchResults> launchesResults) {
         final Statistic statistic = launchesResults.stream()
                 .flatMap(results -> results.getResults().stream())
                 .map(TestResult::getStatus)
                 .collect(Statistic::new, ExtraStatisticMethods::update, ExtraStatisticMethods::merge);
+        final HistoryTrendItem item = new HistoryTrendItem()
+                .withStatistics(statistic)
+                .withExecutorInfo(extractLatestExecutor(launchesResults));
 
-        final List<Statistic> data = launchesResults.stream()
-                .map(results -> results.getExtra(HISTORY_TREND, ArrayList<Statistic>::new))
+        final List<HistoryTrendItem> data = launchesResults.stream()
+                .map(results -> results.getExtra(HISTORY_TREND, ArrayList<HistoryTrendItem>::new))
                 .reduce(new ArrayList<>(), (first, second) -> {
                     first.addAll(second);
                     return first;
                 });
 
         return Stream.concat(
-                Stream.of(statistic),
+                Stream.of(item),
                 data.stream()
         ).limit(20).collect(Collectors.toList());
     }
+
+    private static ExecutorInfo extractLatestExecutor(final List<LaunchResults> launches) {
+        final List<ExecutorInfo> executors = launches.stream()
+                .map(launch -> launch.getExtra("executor"))
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .filter(ExecutorInfo.class::isInstance)
+                .map(ExecutorInfo.class::cast)
+                .collect(Collectors.toList());
+        return executors.isEmpty() ? null : Collections.max(executors,
```

Comment:
you can use sort together with `findFirst` instead and return `Optional`. Then use `getExecutor().ifPresent(item::setExecutorInfo)`